modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpandedComponentAst.java

```diff
@@ -128,6 +139,19 @@ private Object mapErrorMappings(final ComponentParameterAst originalParameter) {
             .getRight();
       }
 
+      private boolean isComplexComponent() {
+        Optional<DslElementSyntax> dslElementSyntax = getDecorated().getGenerationInformation().getSyntax();
+        return dslElementSyntax.map(DslElementSyntax::supportsChildDeclaration).orElse(false);
```

Comment:
Not only verify that this supports a Child declaration but also that is actually is declared as a child.

The difference between this:

    <petstore:config name="petstore" cashierName="jim">
        <petstore:pooled-connection username="#[vars.username]" password="#[vars.password]" />
        <petstore:pets>
            <petstore:pet value="la tota" />
            <petstore:pet value="la porota" />
        </petstore:pets>
    </petstore:config>

and this:

    <petstore:config name="petstore" cashierName="jim">
        <petstore:pooled-connection username="#[vars.username]" password="#[vars.password]" pets="#[vars.pets"]/>
    </petstore:config>